ui/app/controllers/vault/cluster/access/oidc.js

```diff
@@ -1,9 +1,33 @@
 import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
 import { tracked } from '@glimmer/tracking';
 
 export default class OidcConfigureController extends Controller {
+  @service router;
+
   @tracked header = null;
 
+  constructor() {
+    super(...arguments);
+    this.router.on('routeDidChange', (transition) => this.setHeader(transition));
+  }
+
+  setHeader(transition) {
+    // set correct header state based on child route
+    // when no clients have been created, display create button as call to action
+    // list views share the same header with tabs as resource links
+    // the remaining routes are responsible for their own header
+    const routeName = transition.to.name;
```

Comment:
Is this the preferred way to write transitionTo?

Reply:
This is the transition object returned from the `routeDidChange` event on the router service which is different than the `transitionTo` method. On the transition object there are `from` and `to` objects that describe where the transition originated and what the destination is. It's the same object that the `beforeModel` hook receives in routes.